<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>938</width>
    <height>763</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CoreDNS Home</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>corednshome.png</normaloff>corednshome.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Options</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>China DNS Server List</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QListWidget" name="listChinaDNSServers">
           <property name="verticalScrollMode">
            <enum>QAbstractItemView::ScrollPerPixel</enum>
           </property>
           <property name="horizontalScrollMode">
            <enum>QAbstractItemView::ScrollPerPixel</enum>
           </property>
           <property name="uniformItemSizes">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Custom China DNS Servers</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="edtChinaDNSServers">
           <property name="inputMask">
            <string/>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="placeholderText">
            <string>114.114.114.114 223.5.5.5 119.29.29.29 separated by whitespace</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Ads Plugin</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QCheckBox" name="cbAdsPlugin">
           <property name="text">
            <string>Enabled</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QRadioButton" name="rbDefaultAdsPolicy">
             <property name="text">
              <string>Default</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="rbStrictAdsPolicy">
             <property name="text">
              <string>Strict</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>Health Check</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QCheckBox" name="cbHealthCheck">
           <property name="text">
            <string>Enabled</string>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Logs</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QCheckBox" name="cbLogs">
           <property name="text">
            <string>Enabled</string>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>TTL Cache</string>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QCheckBox" name="cbTTLCache">
           <property name="text">
            <string>Enabled</string>
           </property>
          </widget>
         </item>
         <item row="9" column="0">
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>Listen Address</string>
           </property>
          </widget>
         </item>
         <item row="9" column="1">
          <widget class="QLineEdit" name="edtListenAddress">
           <property name="placeholderText">
            <string>:53 or 127.0.0.1:53 or 192.168.1.1:53 etc</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>Custom Ads Black List</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QLineEdit" name="edtAdsBlackList">
           <property name="placeholderText">
            <string>URLs separated by whitespace</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>Ads Policy</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_19">
           <property name="text">
            <string>Google Domain</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QCheckBox" name="cbResolveGoogleDomainByChinaDNS">
           <property name="text">
            <string>Resolved by China DNS server</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Abroad DNS Server List</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QListWidget" name="listAbroadDNSServers">
           <property name="verticalScrollMode">
            <enum>QAbstractItemView::ScrollPerPixel</enum>
           </property>
           <property name="horizontalScrollMode">
            <enum>QAbstractItemView::ScrollPerPixel</enum>
           </property>
           <property name="uniformItemSizes">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Custom Abroad DNS Servers</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="edtAbroadDNSServers">
           <property name="placeholderText">
            <string>8.8.8.8 1.1.1.1 separated by whitespace</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string>Ads List Store</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QCheckBox" name="cbAdsListStore">
           <property name="text">
            <string>Enabled</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>Custom Ads White List</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QLineEdit" name="edtAdsWhiteList">
           <property name="placeholderText">
            <string>URLs separated by whitespace</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_16">
           <property name="text">
            <string>Ads Auto Update Interval</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QLineEdit" name="edtAdsAutoUpdateInterval">
           <property name="placeholderText">
            <string>24h,60m etc, valid time units are &quot;ns&quot;, &quot;us&quot; (or &quot;Âµs&quot;), &quot;ms&quot;, &quot;s&quot;, &quot;m&quot;, &quot;h&quot;. </string>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Configuration Hot Reload</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QCheckBox" name="cbConfigurationHotReload">
           <property name="text">
            <string>Enabled</string>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>System Host File</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QCheckBox" name="cbSystemHostFile">
           <property name="text">
            <string>Enabled</string>
           </property>
          </widget>
         </item>
         <item row="9" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Redis Plugin</string>
           </property>
          </widget>
         </item>
         <item row="9" column="1">
          <widget class="QLineEdit" name="edtRedis">
           <property name="placeholderText">
            <string>127.0.0.1:6379, leave blank if don't need redis cache</string>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="label_18">
           <property name="text">
            <string>Bogus Plugin</string>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QCheckBox" name="cbBogusPlugin">
           <property name="text">
            <string>Enabled</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_20">
           <property name="text">
            <string>Apple Domain</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QCheckBox" name="cbResolveAppleDomainByChinaDNS">
           <property name="text">
            <string>Resolved by China DNS server</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Logs</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QTextEdit" name="edtLogs">
         <property name="undoRedoEnabled">
          <bool>false</bool>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>938</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_Action">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionExportCorefile"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionHomepage"/>
    <addaction name="actionDonateViaPaypal"/>
    <addaction name="actionDonateViaAlipayWechat"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menu_Action_2">
    <property name="title">
     <string>&amp;Action</string>
    </property>
    <addaction name="actionStartCoreDNS"/>
    <addaction name="actionStopCoreDNS"/>
    <addaction name="separator"/>
    <addaction name="actionUpdateCoreDNSBinary"/>
   </widget>
   <addaction name="menu_Action"/>
   <addaction name="menu_Action_2"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionExportCorefile"/>
   <addaction name="separator"/>
   <addaction name="actionStartCoreDNS"/>
   <addaction name="actionStopCoreDNS"/>
   <addaction name="actionUpdateCoreDNSBinary"/>
   <addaction name="separator"/>
   <addaction name="actionDonateViaPaypal"/>
   <addaction name="actionDonateViaAlipayWechat"/>
   <addaction name="actionAbout"/>
  </widget>
  <action name="actionExportCorefile">
   <property name="icon">
    <iconset resource="corednshome.qrc">
     <normaloff>:/res/export.png</normaloff>:/res/export.png</iconset>
   </property>
   <property name="text">
    <string>Export Corefile...</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionHomepage">
   <property name="icon">
    <iconset resource="corednshome.qrc">
     <normaloff>:/res/homepage.png</normaloff>:/res/homepage.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Homepage...</string>
   </property>
  </action>
  <action name="actionDonateViaAlipayWechat">
   <property name="icon">
    <iconset resource="corednshome.qrc">
     <normaloff>:/res/donate.png</normaloff>:/res/donate.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Donate via Alipay/Wechat...</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="corednshome.qrc">
     <normaloff>:/corednshome.png</normaloff>:/corednshome.png</iconset>
   </property>
   <property name="text">
    <string>&amp;About...</string>
   </property>
  </action>
  <action name="actionStartCoreDNS">
   <property name="icon">
    <iconset resource="corednshome.qrc">
     <normaloff>:/res/start.png</normaloff>:/res/start.png</iconset>
   </property>
   <property name="text">
    <string>Start CoreDNS</string>
   </property>
  </action>
  <action name="actionStopCoreDNS">
   <property name="icon">
    <iconset resource="corednshome.qrc">
     <normaloff>:/res/stop.png</normaloff>:/res/stop.png</iconset>
   </property>
   <property name="text">
    <string>Stop CoreDNS</string>
   </property>
  </action>
  <action name="actionUpdateCoreDNSBinary">
   <property name="icon">
    <iconset resource="corednshome.qrc">
     <normaloff>:/res/update.png</normaloff>:/res/update.png</iconset>
   </property>
   <property name="text">
    <string>Update CoreDNS binary</string>
   </property>
  </action>
  <action name="actionDonateViaPaypal">
   <property name="icon">
    <iconset resource="corednshome.qrc">
     <normaloff>:/res/paypal.png</normaloff>:/res/paypal.png</iconset>
   </property>
   <property name="text">
    <string>Donate via Paypal...</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="corednshome.qrc"/>
 </resources>
 <connections/>
</ui>
